core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/file/FileIdentityProvider.java

```diff
@@ -71,7 +80,9 @@ public User auth(String username, String token) {
 			final FileUser user = getFileUser(username);
 			boolean success = user != null && BCrypt.checkpw(token, user.getHashedPassword());
 			if (success) {
-				verifiedTokens.put(username, token);
+				synchronized (verifiedTokens) {
+					verifiedTokens.put(username, token);
```

Comment:
The second parameter will be the user's password according to the superinterface. Is this map meant to cache passwords or tokens?

Reply:
Cache to speed up tests, but it would be great to disable this in PROD. Thoughts?

Reply:
Fully agree 😄

Reply:
Let's raise an issue about this, the FILE provider is not meant to be used in most production environments anyway, so I'm not seeing this as a major issue right now.

Reply:
👍🏻